src/test/api_compatibility.ts

```diff
@@ -14,7 +15,7 @@ export interface ApiCompatibilityChecksOptions {
  * @param {object} createTracer - a factory function that allocates a tracer.
  * @param {object} [options] - the options to be set on api compatibility
  */
-export function apiCompatibilityChecks(createTracer = () => new Tracer(), options: ApiCompatibilityChecksOptions = {}): void {
+function apiCompatibilityChecks(createTracer = () => new Tracer(), options: ApiCompatibilityChecksOptions = {checkInjectBehaviour: true}): void {
```

Comment:
`{checkInjectBehaviour: true}` - this is not sufficient. If someone overrides the Options for another reason, e.g. to skip baggage checks, the default for `checkInjectBehaviour` again reverts to false. That's why it should be a property with a negative name, i.e. `skipXyzChecks`, so that only when someone explicitly opts-in the checks will be skipped.

Same applies to baggage, we should rename it to `skipBaggageChecks`

Reply:
I tried to keep as much as it was originally. I had the same idea, but thought it keep things as they are as much as possible. It seems, however, that this class needs more rework anyways.

Reply:
I'd say if the old way is broken, no reason to continue on the same path.